librz/core/cfile.c

```diff
@@ -53,19 +53,25 @@ static void loadGP(RzCore *core) {
 	}
 }
 
-static RZ_OWN RzList *__save_old_sections(RzCore *core) {
+
+/**
+ * \brief Create a backup of binary's sections.
+ * \param core Rizin core.
+ * \return RzList<RzBinSection>
+ */
+RZ_API RZ_OWN RzList *rz_core_create_sections_backup(RZ_BORROW RzCore *core) {
```

Comment:
Please add

```c
rz_return_val_if_fail(core, NULL)
```

That will return `NULL` and log an error on the console if `core` is `NULL`. You should put it elsewhere as well, wherever you expect the argument to be non-null.